src/renderers/additional/tree-renderer.ts

```diff
@@ -66,28 +66,55 @@ class TreeRenderer extends Renderer implements DataChangeListener {
     const controlElement = <ControlElement> this.uischema;
     const arrayData = this.dataService.getValue(controlElement);
     if (arrayData !== undefined && arrayData.length !== 0) {
-      this.renderDetail(arrayData[0], this.master.lastChild.firstChild);
+      let firstChild = arrayData;
+      if (Array.isArray(firstChild)) {
+        firstChild = firstChild[0];
```

Comment:
Taste of preference, but I'd remove the `firstChild` var and inline the `renderDetail`. Could you add a comment to `this.master.lastChild.firstChild.firstChild`, which looks kinda weird.